sonic-xcvrd/scripts/xcvrd

```diff
@@ -389,6 +389,33 @@ def waiting_time_compensation_with_sleep(time_start, time_to_wait):
     if time_diff < time_to_wait:
         time.sleep(time_to_wait - time_diff)
 
+def wait_for_all_ports_configured():
+    # Connect to APPL_DB and subscribe to PORT table notifications
+    appl_db = swsscommon.DBConnector(swsscommon.APPL_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_MSECS)
+    sel = swsscommon.Select()
+    sst = swsscommon.SubscriberStateTable(appl_db, swsscommon.APP_PORT_TABLE_NAME)
+    sel.addSelectable(sst)
+
+    # Make sure this daemon started after all port configured.
+    while XCVRD_MAIN_TASK_RUNNING_FLAG:
+        (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
+        if state == swsscommon.Select.TIMEOUT:
+            continue
+        if state != swsscommon.Select.OBJECT:
+            log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+            continue
+
+        (key, op, fvp) = sst.pop()
+        if key in ["PortConfigDone", "PortInitDone"]:
+            break
+    # Close the unused socket to prevent mem-leak. Since Xcvrd opens socket to subscribe to
+    # APP_DB's PORT_TABLE events, if Xcvrd doesnot listen on the opened socket, then over a
+    # period of time, redis-DB Send-Q will accumulate events(Eg due to continous port flap)
+    # that can grow up slowly over a period of time resulting in memory leak.
+    sel.removeSelectable(sst)
```

Comment:
>sel.removeSelectable(sst) [](http://example.com/codeflow?start=4&length=25)

Is it the most critical fix?
I am considering enforce it to all the use case by default.

Reply:
The critical one is the ```sel``` object at line 398 which gets deleted after the function completes which results in socket getting closed and the Send-Q buffer cleared. But it's good to call removeSelectable() as a clean up before object gets deleted.